src/gui/sdlmain.cpp

```diff
@@ -1308,38 +1391,48 @@ static SDL_Window *SetupWindowScaled(SCREEN_TYPES screen_type, bool resizable)
 	if (sdl.scaling_mode == SCALING_MODE::PERFECT)
 		return setup_window_pp(screen_type, resizable);
 
-	int w, h;
+	int window_width;
+	int window_height;
 	if (sdl.desktop.fullscreen) {
-		w = sdl.desktop.full.fixed ? sdl.desktop.full.width : 0;
-		h = sdl.desktop.full.fixed ? sdl.desktop.full.height : 0;
+		window_width = sdl.desktop.full.fixed ? sdl.desktop.full.width : 0;
+		window_height = sdl.desktop.full.fixed ? sdl.desktop.full.height : 0;
 	} else {
-		w = sdl.desktop.window.width;
-		h = sdl.desktop.window.height;
+		window_width  = sdl.desktop.window.width;
+		window_height = sdl.desktop.window.height;
 	}
 
-	if (w > 0 && h > 0) {
-		sdl.clip = calc_viewport_fit(w, h);
-		sdl.window = SetWindowMode(screen_type, w, h,
-		                           sdl.desktop.fullscreen, resizable);
+	if (window_width > 0 && window_height > 0) {
+		sdl.window = SetWindowMode(screen_type,
+		                           window_width,
+		                           window_height,
+		                           sdl.desktop.fullscreen,
+		                           resizable);
+
+		const auto canvas = get_canvas_size(screen_type);
+		assert(canvas.w > 0 && canvas.h > 0);
```

Comment:
(can drop assert here with `get_canvas_size` checking for us)

Reply:
Done.